core/src/main/java/tech/tablesaw/io/DataFrameReader.java

```diff
@@ -56,14 +56,13 @@ public Table url(String url) {
    * mime-type Use {@link #usingOptions(ReadOptions) usingOptions} to use non-default options
    */
   public Table url(URL url) {
-    URLConnection connection = null;
     try {
-      connection = url.openConnection();
+      URLConnection connection = url.openConnection();
+      String contentType = connection.getContentType();
+      return url(url, getCharset(contentType), getMimeType(contentType));
     } catch (IOException e) {
-      e.printStackTrace();
+        throw new RuntimeIOException(e);
```

Comment:
we should probably setup spotless to catch stuff like this
```suggestion
      throw new RuntimeIOException(e);
```

Reply:
I thought there was a commit hook for that